### PR TITLE
unified headers fix (master)

### PR DESCRIPTION
**Pull request by rraallvv**
Enable using NDK unified headers when available and opt-out ([more info](https://android.googlesource.com/platform/ndk.git/+/master/docs/Roadmap.md))

### CODE REVIEW

**Comment by akien-mga** [comment]
I still don't see why we need an option at all. It doesn't matter to use whether we use unified headers or not, the result is the same and we just want it to build. Having the option makes sense for very specific Android apps, but in our case it's "one size fits all" and such customization isn't relevant for game devs.

**Comment by rraallvv** [comment]
@akien-mga I see... I removed the option `ndk_unified_headers`, but I'm not sure if the helper functions should be squashed in one function (`get_ndk_version(), same_version_or_newer(), and have_unified_headers()`), or if the detection logic should be added in the body of `configure()`.

File: platform/android/detect.py [hunk 1]
```diff
@@ -179,26 +178,22 @@ def mySpawn(sh, escape, cmd, args, env):
...
-        ndk_unified_headers = True
-        print("Turning NDK unified headers on (starting from r16)")
-
     lib_sysroot = env["ANDROID_NDK_ROOT"] + "/platforms/" + env['ndk_platform'] + "/" + env['ARCH']
 
     ## Compile flags
 
-    if ndk_unified_headers:
+    ndk_version = get_ndk_version(env["ANDROID_NDK_ROOT"])
+    if ndk_version != None and LooseVersion(ndk_version) >= LooseVersion("15.0.4075724"):
```
Comment (akien-mga) [inline platform/android/detect.py]: Well I merged too fast, this commit is incomplete. In the 2.1 one you have `from distutils.version import LooseVersion`, it's missing here.

Comment (rraallvv) [reply platform/android/detect.py]: @akien-mga My bad, somehow the built didn't fail alerting of the missing import. [This PR](https://github.com/godotengine/godot/pull/12782) has it. Sorry about that.

**Pull request merged**